### ISSUE TITLE
[page types] Define page types for Add-ons

### ISSUE THREAD

**Issue by wbamberg** [mdn/content#23942]
Part of https://github.com/mdn/content/issues/15539.

Some context, for people not familiar. This is part of a project to add a front matter item to every MDN page, indicating what kind of page it is (and/or what kind of feature it describes). This is going to be useful anywhere we want to identify the type of a page with any kind of precision, for example to check that pages follow a consistent structure, or to list pages in the right place in sidebars. We've so far added page type values for most of the big areas of MDN, including [Web/API](https://github.com/mdn/content/issues/16255), [JavaScript](https://github.com/mdn/content/issues/16156), [CSS](https://github.com/mdn/content/issues/15540), [HTML](https://github.com/mdn/content/issues/22705), and [SVG](https://github.com/mdn/content/issues/22762).

This issue is to add page types to pages in the [Add-ons](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons) part of MDN and in particular WebExtensions.

The general rules for page types are: whenever we might want to make a distinction between different sorts of pages (for example, because they need to have a different structure, or be grouped differently in a sidebar or some other piece of nav), then we should consider defining page types to distinguish them.

I've trawled through the pages and tentatively assigned page types to them, in the spreadsheet: https://docs.google.com/spreadsheets/d/14RX8EEKPpeYkP2cF5Y_ZKF3JpNJ_24OL7stvZWNJLIY/edit#gid=2054492661.

Apart from the generic page types `guide` and `landing-page`, I've come up with the following types for Add-ons:

* `webextension-api` - for things like [`alarms`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/alarms)
* `webextension-api-event` - for things like [`action.onClicked`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/webextensions/api/action/onclicked)
* `webextension-api-function` - for things like [`action.setBadgeText()`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/webextensions/api/action/setbadgetext)
* `webextension-api-property` - for things like [`browserSettings.openBookmarksInNewTabs`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/webextensions/api/browsersettings/openbookmarksinnewtabs)
* `webextension-api-type` -  for things like [`contextualIdentities.ContextualIdentity`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/webextensions/api/contextualidentities/contextualidentity)
* `webextension-manifest-key` -  for things like [`user_scripts`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/webextensions/manifest.json/user_scripts)

@rebloor , I'd like to know whether you think this is a sensible taxonomy of the WebExtensions docs. If you agree I can file some PRs to add the front matter key.

**Comment by wbamberg** [comment]
> @wbamberg sorry for the delay in getting back to you. The list looks good. However, if page content and sidebar grouping are reasons for creating types, should we consider:
> 
>     * `webextension-concept`
> 
>     * `webextension-userinterface`
> 
>     * `webextension-howto`
> 
> 
> as well?

This is somewhat up to you, but I would not do this, for now.

So far I've tried to be quite conservative with page types, and have just a couple of page types that are outside "reference documentation" - these two are `landing-page` and `guide`. We might decide at some point that a howto, for example, is a distinct sort of page from a normal guide page, but if we did I think that would be a generic type as well - that is, a WebExtensions howto would be the same sort of thing as a CSS howto. For now I think we should be reasonably minimalist and call them all guides, and we can always go and make finer distinctions later if we think they will have a practical use.

My fear is that if we apply this sort of granularity all across MDN we will end up with hundreds of types.
